File: example-defold/main/wavedash.lua
local M = {}

local function run(code)
    if html5 then html5.run(code) end
end

function M.init()
    run("window.WavedashJS && window.WavedashJS.init({ debug: true })")
end

function M.update_load_progress(fraction)
    run("window.WavedashJS && window.WavedashJS.updateLoadProgressZeroToOne(" .. tostring(fraction) .. ")")
end

return M

File: example-rpgmaker/js/plugins/Wavedash.js
//=============================================================================
// Wavedash SDK
//=============================================================================

/*:
 * @target MZ
 * @plugindesc Initializes the Wavedash SDK and enables canvas stretch-to-fit in web browsers.
 * @author Wavedash
 */

(() => {
    // Force Graphics._stretchEnabled = true in web browsers. RPG Maker MZ only
    // stretches the canvas for native (nwjs) or mobile by default, which leaves
    // desktop-web builds locked at the native 816x624 resolution regardless of
    // viewport size. Overriding _defaultStretchMode makes the canvas scale to
    // the window, preserving aspect ratio.
    const _Graphics_defaultStretchMode = Graphics._defaultStretchMode;
    Graphics._defaultStretchMode = function() {
        return true;
    };

    const _Scene_Boot_start = Scene_Boot.prototype.start;
    Scene_Boot.prototype.start = function() {
        _Scene_Boot_start.call(this);
        (async () => {
            try {
                const WavedashJS = await window.WavedashJS;
                WavedashJS.updateLoadProgressZeroToOne(1);
                WavedashJS.init({
                    debug: true
                });
            } catch (e) {
                console.warn("[wavedash] init failed:", e);
            }
        })();
    };
})();
